macos/FluentUI/Button.swift

```diff
@@ -237,6 +261,109 @@ open class Button: NSButton {
 		}
 	}
 	
+	
+	class FluentButtonCell : NSButtonCell {
+		override func imageRect(forBounds rect: NSRect) -> NSRect {
+			let isRTL = NSApp.userInterfaceLayoutDirection == .rightToLeft
+			let imageRect = super.imageRect(forBounds: rect)
+			let titleSize = title.size(withAttributes: [.font: font as Any])
+			
+			var x = imageRect.origin.x
+			var y = imageRect.origin.y
+			var width = imageRect.size.width
+			var height = imageRect.size.height
+			
+			switch imagePosition {
+			case .noImage:
+				x = 0
+				y = 0
+				width = 0
+				height = 0
+				break
+			case .imageOnly:
+				x = Button.horizontalEdgePadding
+				y = Button.verticalEdgePadding
+				width = rect.width - (Button.horizontalEdgePadding * 2)
+				height = rect.height - (Button.verticalEdgePadding * 2)
+			case .imageLeading:
+				x = isRTL ? titleSize.width + Button.titleAndImageRectInterspacing : 0
+				y = Button.verticalEdgePadding
+				width = rect.width - Button.titleAndImageRectInterspacing - titleSize.width
+				height = rect.height - (Button.verticalEdgePadding * 2)
+				break
+			case .imageTrailing:
+				x = isRTL ? 0 : titleSize.width + Button.titleAndImageRectInterspacing
+				y = Button.verticalEdgePadding
+				width = rect.width - Button.titleAndImageRectInterspacing - titleSize.width
+				height = rect.height - (Button.verticalEdgePadding * 2)
+				break
+			default:
+				// We explicitly do not support the other variants of imagePosition
+				break
+			}
+			
+			return NSMakeRect(x, y, width, height)
+		}
+		
+		override func titleRect(forBounds rect: NSRect) -> NSRect {
+			let isRTL = NSApp.userInterfaceLayoutDirection == .rightToLeft
+			let titleRect = super.titleRect(forBounds: rect);
+			let titleSize = title.size(withAttributes: [.font: font as Any])
+			
+			var x = titleRect.origin.x
+			var y = titleRect.origin.y
+			var width = titleRect.size.width
+			var height = titleRect.size.height
+			
+			let imageSize = image?.size ?? NSZeroSize
+						
+			switch imagePosition {
+			case .noImage:
+				x = 0
+				y = 0
+				width = titleRect.width + (Button.horizontalEdgePadding * 2)
+				height = titleRect.height + (Button.verticalEdgePadding * 2)
+				break
+			case .imageOnly:
+				x = 0
+				y = 0
+				width = 0
+				height = 0
+				break
+			case .imageLeading:
+				x = isRTL ? Button.horizontalEdgePadding : Button.horizontalEdgePadding + imageSize.width + Button.titleAndImageRectInterspacing
+				y = Button.verticalEdgePadding
+				width = titleSize.width
+				height = titleSize.height
+				break
+			case .imageTrailing:
+				x = isRTL ? Button.horizontalEdgePadding + imageSize.width + Button.titleAndImageRectInterspacing : Button.horizontalEdgePadding
+				y = Button.verticalEdgePadding
+				width = titleSize.width
+				height = titleSize.height
+				break
+			default:
+				// We explicitly do not support the other variants of imagePosition
+				break
+			}
```

Comment:
Could also put a preconditionFailure here or something. As this flow goes, it'll follow whatever super returns, which'll probably always look bad

Reply:
i prefer not having default so ppl dont make mistakes

Reply:
That would mean implementing the imagePositions like "imageAbove, below, overlaps"... which feel weird. Though I guess I've already done the math and can..

Reply:
... I went for it. This also adds an "@unknown default" case bc apparently swift wants that